### PR TITLE
feat: removed API roles from routes, updated tests

### PR DESCRIPTION
**Pull request by aws-tyler**
Description of changes:
Removed `PUT` /roles/:roleName and `POST` /roles and updated the associated integration tests.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/support/utils/settings.ts [hunk 1]
```diff
@@ -19,7 +19,6 @@ interface Setting {
   projectAdmin1PasswordParamStorePath: string;
   projectAdmin2UserNameParamStorePath: string;
   projectAdmin2PasswordParamStorePath: string;
-  projectForRolesTesting: string;
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/support/utils/settings.ts:22]: This is an old setting I found while doing this PR that is no longer needed.

File: workbench-core/example/infrastructure/integration-tests/tests/users/getUserRoles.test.ts [hunk 2]
```diff
@@ -36,14 +36,9 @@ describe('userManagement get user roles integration test', () => {
...
 
   it('returns a list containing the users roles', async () => {
-    const roleName = `example-role-name-${uuidv4()}`; // TODO use roles.create() response once the route returns the created role name
     const { data: userData } = await adminSession.resources.users.create(user);
-    await adminSession.resources.roles.create({ roleName });
-    await adminSession.resources.roles.role(roleName).addUser({ userId: userData.id });
-
     const { data } = await adminSession.resources.users.user(userData.id).getRoles();
-
-    expect(data).toMatchObject({});
```
Comment (DaveMadsen) [inline workbench-core/example/infrastructure/integration-tests/tests/users/getUserRoles.test.ts:46]: why remove the UserManagement component test?

Comment (aws-tyler) [reply workbench-core/example/infrastructure/integration-tests/tests/users/getUserRoles.test.ts:46]: I completely forgot a separate router is used in the example app :facepalm:. Replacing that now (but will keep the expectation change).

**Review by DaveMadsen — APPROVED** [review]
thinking it would have been better to correct the `example` integration test by matching an array that contains `roleName`, but what you have here should work.

**Review by bhudnell — DISMISSED** [review]

File: solutions/swb-app/src/userRoutes.ts [hunk 3]
```diff
@@ -157,29 +157,4 @@ export function setUpUserRoutes(router: Router, userService: UserManagementServi
...
-      const validatedRequest = validateAndParse<CreateRoleRequest>(CreateRoleRequestParser, req.body);
-      const response = await userService.createRole(validatedRequest.roleName);
-      res.status(201).send(response);
-    })
-  );
-
-  router.put(
-    '/roles/:roleName',
-    wrapAsync(async (req: Request, res: Response) => {
-      const validatedRequest = validateAndParse<UpdateRoleRequest>(UpdateRoleRequestParser, {
```
Comment (maghirardelli) [inline solutions/swb-app/src/userRoutes.ts:173]: What about cleaning up `UpdateRoleRequest`?

**Pull request closed without merging**